### PR TITLE
EDC-21 Add functionality users(Teachers, Students) and DB migration

### PR DESCRIPTION
**Pull request by Natanius18**

### CODE REVIEW

File: src/main/java/software/sigma/internship/controller/RestStudentController.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import software.sigma.internship.validator.exception.UserNotFoundException;
+import software.sigma.internship.entity.user.Student;
+
+import javax.validation.Valid;
+import java.util.List;
+
+@RestController
+@AllArgsConstructor
+@RequestMapping(path = "/v1/students")
+public class RestStudentController {
```
Comment (PoletaevaSasha) [inline src/main/java/software/sigma/internship/controller/RestStudentController.java]: don't need Rest in controller name

File: build.gradle [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+}
+
+dependencies {
+	implementation 'org.springframework.boot:spring-boot-starter'
+	implementation 'org.springframework.boot:spring-boot-starter-web'
+	implementation 'org.jetbrains:annotations:23.0.0'
+	compileOnly 'org.projectlombok:lombok'
+	annotationProcessor 'org.projectlombok:lombok'
+	testImplementation 'org.springframework.boot:spring-boot-starter-test'
+	implementation 'org.jdbi:jdbi3-sqlobject:3.28.0'
```
Comment (PoletaevaSasha) [inline build.gradle]: please remove unused dependency

File: src/main/resources/db/migration/V1__init.sql [hunk 3]
```diff
@@ -1,9 +0,0 @@
-CREATE TABLE person
```
Comment (PoletaevaSasha) [inline src/main/resources/db/migration/V1__init.sql]: Why this file is empty? I am not sure, that you need V2 of this file.

File: src/main/java/software/sigma/internship/entity/Teacher.java [hunk 4]
```diff
@@ -1,12 +1,11 @@
...
 
 import lombok.AllArgsConstructor;
 import lombok.Builder;
 import lombok.EqualsAndHashCode;
 import lombok.Getter;
 import lombok.NoArgsConstructor;
 import lombok.Setter;
-import software.sigma.internship.entity.additional.Position;
 
 import javax.persistence.*;
```
Comment (PoletaevaSasha) [inline src/main/java/software/sigma/internship/entity/Teacher.java]: you still have import *  :)

File: src/main/java/software/sigma/internship/entity/Position.java [hunk 5]
```diff
@@ -1,4 +1,4 @@
-package software.sigma.internship.entity.additional;
+package software.sigma.internship.entity;
 
 public enum Position {
```
Comment (PoletaevaSasha) [inline src/main/java/software/sigma/internship/entity/Position.java]: Please move to enum package

Comment (Natanius18) [reply src/main/java/software/sigma/internship/entity/Position.java]: I've added as inner enum in class Teacher

**Comment by PoletaevaSasha** [comment]
Looks good. You can mere changes.

**Pull request merged**